config.toml

```diff
@@ -0,0 +1,221 @@
+encode_columns = ['Billing Account Number', 'Billing Account Name', 'Wireless Number', 'User Name', 'FAN Invoice Number', 'Data Pooling Rate Plan Code', 'Data Pool Name']
+# Encode value in `Section_3` if `Section_2` contains value `User Name`. We actually need the remaining values.
+encode_conditional = [['Section_3', 'Section_2', '==', 'User Name']]
+
+
+[Rogers]
+[Rogers.common]
+carrier = 'Rogers'
+config_class = 'csv-config'
+delimiter = '|'
+encoding = 'iso-8859-1'
+
+[Rogers.Custom]
+skip_initial_lines = 2
+file_mask = 'ALL_CALLS-Custom'
+clear_columns = ['Report Level']
+encode_columns = ['Billing Account', 'User Number', 'User Name']
+
+[Rogers.GPRS]
+file_mask = 'ALL_CALLS-GPRS\.'
+clear_columns = []
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+# This file will be searched in a path relative to the data that was found.
+external_header_file = 'Header-GPRS.txt'
+
+[Rogers.GPRS_RM]
+file_mask = 'ALL_CALLS-GPRS-Rm\.'
+# Despite nice information coming from Carrier Name, it also contains User Number.
+clear_columns = []
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-GPRS.txt'
+# Encode each named group in given column according to the regular expression (in Python `re` format).
+# In current form it doesn't work with anything that's escaped, so no `\w` etc.
+# In this case, "Carrier Name" contains the "User Number" at the very end. The same issue is not present in Voice_RM.
+encode_regex = [['^.* (?P<user_number>[0-9]+)$', 'Carrier Name']]
+
+[Rogers.SMS]
+file_mask = 'ALL_CALLS-SMS\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-SMS.txt'
+# Encode value in `Called Number` if it's not an asterisk.
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+
+[Rogers.Voice]
+file_mask = 'ALL_CALLS-Voice\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-Voice.txt'
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+
+[Rogers.Voice_RM]
+file_mask = 'ALL_CALLS-VoiceRM\.'
+clear_columns = ['Subscriber ID 1', 'Subscriber ID 2', 'Subscriber ID 3', 'Subscriber ID 4']
+encode_columns = ['BAN', 'Invoice No.', 'User Name', 'User Number']
+external_header_file = 'Header-Voice.txt'
+encode_conditional = [['Called Number', 'Called Number', '!=', '*']]
+# Note that GPRS_RM file has this issue where "User Number" is placed in "Carrier Name" field. This issue
+# is not present in the Voice_RM file, in which only the country and the carrier are listed.
+
+
+[Telus]
+[Telus.common]
+carrier = 'Telus'
+config_class = 'raw-regex-config'
+encoding = 'iso-8859-1'
```

Comment:
to me it kinda seems like it would be best to treat most files as binary and do subsitutions on binary data, that way we wouldn't even need to consider encoding